vms/proposervm/state_syncable_vm_test.go

```diff
@@ -590,3 +594,134 @@ func TestStateSummaryAcceptOlderBlock(t *testing.T) {
+			IDV: ids.GenerateTestID(),
+		},
+		BytesV:  []byte{1},
+		ParentV: ids.GenerateTestID(),
+		HeightV: 1969,
+	}
+	innerBlk2 := snowmantest.BuildChild(innerBlk1)
+
+	innerSummary1 := &blocktest.StateSummary{
+		IDV:     innerBlk1.ID(),
+		HeightV: innerBlk1.Height(),
+		BytesV:  innerBlk1.BytesV,
+	}
+
+	require.NoError(vm.SetForkHeight(innerSummary1.Height() - 1))
+
+	// Set the last accepted block height to be higher that the state summary
+	// we are going to attempt to accept
+	vm.lastAcceptedHeight = innerBlk2.Height()
+
+	innerVM.LastAcceptedF = func(context.Context) (ids.ID, error) {
+		return innerBlk2.IDV, nil
+	}
+
+	innerVM.GetBlockF = func(_ context.Context, blkID ids.ID) (snowman.Block, error) {
+		switch blkID {
+		case snowmantest.GenesisID:
+			return snowmantest.Genesis, nil
+		case innerBlk1.ID():
+			return innerBlk1, nil
+		case innerBlk2.ID():
+			return innerBlk2, nil
+		default:
+			return nil, database.ErrNotFound
+		}
+	}
+	innerVM.GetStateSummaryF = func(context.Context, uint64) (block.StateSummary, error) {
+		return innerSummary1, nil
+	}
+	innerVM.ParseBlockF = func(_ context.Context, b []byte) (snowman.Block, error) {
+		switch {
+		case bytes.Equal(b, innerBlk1.BytesV):
+			return innerBlk1, nil
+		case bytes.Equal(b, innerBlk2.BytesV):
+			return innerBlk2, nil
+		default:
+			panic("unexpected parse block")
+		}
+	}
+	// Setup innerSummary.AcceptF to return StateSyncSkipped and check it was called
+	calledInnerAccept := false
+	innerSummary1.AcceptF = func(context.Context) (block.StateSyncMode, error) {
+		calledInnerAccept = true
+		return block.StateSyncSkipped, nil
+	}
+
+	slb1, err := statelessblock.Build(
+		vm.preferred,
+		innerBlk1.Timestamp(),
+		100, // pChainHeight,
+		vm.StakingCertLeaf,
+		innerBlk1.Bytes(),
+		vm.ctx.ChainID,
+		vm.StakingLeafSigner,
+	)
+	require.NoError(err)
+	proBlk1 := &postForkBlock{
+		SignedBlock: slb1,
+		postForkCommonComponents: postForkCommonComponents{
+			vm:       vm,
+			innerBlk: innerBlk1,
+		},
+	}
+	require.NoError(vm.acceptPostForkBlock(proBlk1))
+
+	slb2, err := statelessblock.Build(
+		vm.preferred,
+		innerBlk2.Timestamp(),
+		100, // pChainHeight,
+		vm.StakingCertLeaf,
+		innerBlk2.Bytes(),
+		vm.ctx.ChainID,
+		vm.StakingLeafSigner,
+	)
+	require.NoError(err)
+	proBlk2 := &postForkBlock{
+		SignedBlock: slb2,
+		postForkCommonComponents: postForkCommonComponents{
+			vm:       vm,
+			innerBlk: innerBlk2,
+		},
+	}
+	require.NoError(vm.acceptPostForkBlock(proBlk2))
+
+	summary, err := vm.GetStateSummary(context.Background(), innerBlk1.Height())
+	require.NoError(err)
+	require.Equal(summary.Height(), innerBlk1.Height())
+
+	// Process a state summary that would rewind the chain
+	// ProposerVM should ignore the rollback and accept the inner state summary to
+	// notify the innerVM.
+	// This can result in the ProposerVM and innerVM diverging their last accepted block,
+	// which should be rolled forward and re-aligned in SetState.
```

Comment:
We do not roll forward. If the innerVM rolls back, the proposervm will roll back. This PR is ensuring that the proposervm _only_ rolls back if the innerVM rolls back.